HW2/util/set_compiler.py

```diff
@@ -6,11 +6,11 @@ def install():
         # most OSX machines now use clang-based compilers.
         # clang only recently got support for OpenMP.
         # search for a good compiler
-        if os.path.exists('/usr/local/bin/gcc'):
-            os.environ['CC'] = '/usr/local/bin/gcc'
-            print("Compiling with /usr/local/bin/gcc")
-        elif os.path.exists('/usr/local/bin/clang-omp'):
+        if os.path.exists('/usr/local/bin/clang-omp'):
             os.environ['CC'] = '/usr/local/bin/clang-omp'
             print("Compiling with /usr/local/bin/clang-omp")
+        elif os.path.exists('/usr/local/bin/gcc'):
+            os.environ['CC'] = '/usr/local/bin/gcc'
+            print("Compiling with /usr/local/bin/gcc")
         else:
             print("No definitely good compiler found, things may not work.")
```

Comment:
Very interesting to see your way of coding!
It gives me some new insights, and it is very efficient.
I think it is a good job over all!